osu.Game/Beatmaps/BeatmapUpdaterMetadataLookup.cs

```diff
@@ -89,7 +89,16 @@ private void lookup(BeatmapSetInfo set, BeatmapInfo beatmapInfo, bool preferOnli
 
                 if (res != null)
                 {
-                    beatmapInfo.Status = res.Status;
+                    beatmapInfo.OnlineID = res.OnlineID;
+                    beatmapInfo.OnlineMD5Hash = res.MD5Hash;
+                    beatmapInfo.LastOnlineUpdate = res.LastUpdated;
+                    beatmapInfo.Metadata.Author.OnlineID = res.AuthorID;
+
+                    // In the case local changes have been applied, don't reset the status.
+                    if (beatmapInfo.MatchesOnlineVersion || beatmapInfo.Status != BeatmapOnlineStatus.LocallyModified)
```

Comment:
I guess there’s an argument to be made for that (would a user expect the pill to only change from “pending” to “ranked” after first applying an update?). Although this conditional would need to be changed to allow status upgrades from `None` at least.

Reply:
Not sure I follow the above reply, but maybe it's something I'm not seeing? The conditional would allow changes from `None`, I think? Status would then update if the md5 hashes match *and* the beatmap hasn't been previously touched locally, rather than either-or as written...?

Reply:
Yeah, but a user would want to probably see the pill show even before an update is applied. Dunno. If we go with the proposed conditional then this whole change needs to be rethought (the status text should read differently and be shown in cases an update is required, inside the pill, as opposed to nothing showing unless hashes match).

Reply:
I think I see it now. The mistake I made is that I assumed this method would only ever fire post-actual-local-update, but I see now that it can also fire if an online change is detected.